Google.PowerShell.IntegrationTests/Container/Container.GkeCluster.Tests.ps1

```diff
@@ -186,81 +187,98 @@ Describe "New-GkeNodeConfig" {
     }
 }
 
+# Given a network name, create a network name and extract
+# out a subnetwork that corresponds to region $region.
+function New-NetworkAndSubnetwork($networkName, $region) {
+    gcloud compute networks create $networkName 2>$null | Out-Null
+    $network = Get-GceNetwork $networkName
+    $subnet = $network.Subnetworks | Where-Object {$_.Contains($region)}
+    $subnet -match "subnetworks/([^/]*)" | Out-Null
+    return $Matches[1]
+}
+
+# Job for creating a cluster.
+$script:clusterCreationWithoutUsingNodeConfigScriptBlock = {
+    param($cmdletPath, $addGkeParameters)
+
+    # Create a credential if that is provided.
+    if ($null -ne $addGkeParameters["clusterUserName"]-and
+        $null -ne $addGkeParameters["clusterPassword"]) {
+        $password = ConvertTo-SecureString $addGkeParameters["clusterPassword"] -AsPlainText -Force
+        $credential = New-Object System.Management.Automation.PSCredential `
+                                ($addGkeParameters["clusterUserName"], $password)
+        $addGkeParameters["MasterCredential"] = $credential
+        $addGkeParameters.Remove("clusterUserName")
+        $addGkeParameters.Remove("clusterPassword")
+    }
+
+    . $cmdletPath
+    Install-GCloudCmdlets | Out-Null
+    $PSBoundParameters.Remove("cmdletPath")
+    Add-GkeCluster @addGkeParameters
+}
+
 Describe "Add-GkeCluster" {
     $r = Get-Random
```

Comment:
Given how long this takes I think its probably the best for now.  Can you make a explicit comment and call this out and add why it's done this way?